server/go/subscription/manager.go

```diff
@@ -0,0 +1,67 @@
+package subscription
+
+import "sync"
+
+type Manager struct {
+	subscriptions map[string][]Subscriber
+	mutex         sync.Mutex
+}
+
+var managerInstance *Manager = nil
+
+func GetManager() *Manager {
```

Comment:
I don't think it's a great idea to have a singleton. It would be better to have a `subscription.NewManager()` func, and pass that around as a dependency. We could even export a `Manager` interface (and have the struct unexported) so that the code depending on this is not coupled to the actual implementation.